Tests/Tests/STPPaymentConfigurationTest.m

```diff
@@ -0,0 +1,120 @@
+}
+
+- (void)testInit {
+    STPPaymentConfiguration *paymentConfiguration = [[STPPaymentConfiguration alloc] init];
+
+    XCTAssertNil(paymentConfiguration.publishableKey);
+    XCTAssertEqual(paymentConfiguration.additionalPaymentMethods, STPPaymentMethodTypeAll);
+    XCTAssertEqual(paymentConfiguration.requiredBillingAddressFields, STPBillingAddressFieldsNone);
+    XCTAssertEqual(paymentConfiguration.requiredShippingAddressFields, PKAddressFieldNone);
+    XCTAssert(paymentConfiguration.verifyPrefilledShippingAddress);
+    XCTAssertEqual(paymentConfiguration.shippingType, STPShippingTypeShipping);
+    XCTAssertNil(paymentConfiguration.companyName); // Defaults to nil for test target
+    XCTAssertNil(paymentConfiguration.appleMerchantIdentifier);
+    XCTAssert(paymentConfiguration.canDeletePaymentMethods);
+}
+
+- (void)testApplePayEnabledSatisfied {
+    id stripeMock = OCMClassMock([Stripe class]);
+    OCMStub([stripeMock deviceSupportsApplePay]).andReturn(YES);
+
+    STPPaymentConfiguration *paymentConfiguration = [[STPPaymentConfiguration alloc] init];
+    paymentConfiguration.appleMerchantIdentifier = @"appleMerchantIdentifier";
+    paymentConfiguration.additionalPaymentMethods = STPPaymentMethodTypeAll;
+
+    XCTAssert([paymentConfiguration applePayEnabled]);
+}
+
+- (void)testApplePayEnabledMissingAppleMerchantIdentifier {
+    id stripeMock = OCMClassMock([Stripe class]);
+    OCMStub([stripeMock deviceSupportsApplePay]).andReturn(YES);
+
+    STPPaymentConfiguration *paymentConfiguration = [[STPPaymentConfiguration alloc] init];
+    paymentConfiguration.appleMerchantIdentifier = nil;
+    paymentConfiguration.additionalPaymentMethods = STPPaymentMethodTypeAll;
+
+    XCTAssertFalse([paymentConfiguration applePayEnabled]);
+}
+
+- (void)testApplePayEnabledDisallowAdditionalPaymentMethods {
+    id stripeMock = OCMClassMock([Stripe class]);
+    OCMStub([stripeMock deviceSupportsApplePay]).andReturn(YES);
+
+    STPPaymentConfiguration *paymentConfiguration = [[STPPaymentConfiguration alloc] init];
+    paymentConfiguration.appleMerchantIdentifier = @"appleMerchantIdentifier";
+    paymentConfiguration.additionalPaymentMethods = STPPaymentMethodTypeNone;
+
+    XCTAssertFalse([paymentConfiguration applePayEnabled]);
+}
+
+- (void)testApplePayEnabledMisisngDeviceSupport {
+    id stripeMock = OCMClassMock([Stripe class]);
+    OCMStub([stripeMock deviceSupportsApplePay]).andReturn(NO);
+
+    STPPaymentConfiguration *paymentConfiguration = [[STPPaymentConfiguration alloc] init];
+    paymentConfiguration.appleMerchantIdentifier = @"appleMerchantIdentifier";
+    paymentConfiguration.additionalPaymentMethods = STPPaymentMethodTypeAll;
+
+    XCTAssertFalse([paymentConfiguration applePayEnabled]);
+}
+
+#pragma mark - Description
+
+- (void)testDescription {
+    STPPaymentConfiguration *paymentConfiguration = [[STPPaymentConfiguration alloc] init];
+    XCTAssert(paymentConfiguration.description);
```

Comment:
I just tried going down the path of `class_copyPropertyList` and it "works" but it's kind of sketchy because we have to:
- Do a bunch of runtime lookups (probably okay, just some ugly code)
- Derive from `class_copyIvarList` which is closer to what we log but then we go into matching implementation specifics
- Subtract any properties introduced by protocols with `class_copyProtocolList` and `protocol_copyPropertyList`
- Subtract any properties we intentionally don't log because of PII
- Add any derivations like logging `last4` instead of `accountNumber`

So in the end, we'll end up with a test that's going to fail more often than it should because of unexpected reasons.